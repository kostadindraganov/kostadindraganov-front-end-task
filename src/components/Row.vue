<template>
  <div class="time-row relative w-100" ref="row">
    <div class="absolute -left-20 -top-33">
      <div class="w-16 text-right">
      <span class="text-lg text-slate-500"> {{title}} </span>

      </div>
    </div>

    <div v-if="showRow" class="time-body bg-gray-100 h-15.5 border-l-2 border-l-slate-200 box-border">

    </div>
  
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class Row extends Vue {
  @Prop() private title!: string;
  @Prop({ 
    type: Boolean, 
    default: false 
  }) private isLast!: boolean;

  get showRow(): boolean {
    return !this.isLast;
  }

}
</script>
