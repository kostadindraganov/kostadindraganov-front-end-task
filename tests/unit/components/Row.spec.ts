import { shallowMount } from '@vue/test-utils';
import Row from '@/components/Row.vue';

let wrapper;

describe('Testing component Row', () => {
	describe('Row', () => {
        wrapper = shallowMount(Row, {
            propsData: {
                title: 'test',
                isLast: false
            }
        })

        it('Row is a Vue instance', () => {
            expect(wrapper.vm).toBeTruthy()
        })
        
        it('Row has proper layout', () => {
            expect(wrapper.html()).toMatchSnapshot()
        })

        it('Test prop isLast', () => {
            wrapper.setProps({ isLast: true })
            const body = wrapper.find('.time-body');
            expect(body).toBeTruthy()
        })


        it('Test prop title', () => {
            wrapper.setProps({ isLast: true })
            expect(wrapper.vm.title).toBe('test')
        })
	});
});