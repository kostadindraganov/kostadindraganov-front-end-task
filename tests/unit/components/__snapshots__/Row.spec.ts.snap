// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component Row Row Row has proper layout 1`] = `
<div class="time-row relative w-100">
  <div class="absolute -left-20 -top-33">
    <div class="w-16 text-right"><span class="text-lg text-slate-500"> test </span></div>
  </div>
  <div class="time-body bg-gray-100 h-15.5 border-l-2 border-l-slate-200 box-border"></div>
</div>
`;
